substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1283,10 +1285,18 @@ pub mod pallet {
 			payee: RewardDestination<T::AccountId>,
 		) -> DispatchResult {
 			let controller = ensure_signed(origin)?;
-			let ledger = Self::ledger(Controller(controller))?;
+			let ledger = Self::ledger(Controller(controller.clone()))?;
+
+			let controller_deprecated = || -> RewardDestination<T::AccountId> {
+				#[allow(deprecated)]
+				RewardDestination::Controller
+			};
+
+			ensure!(payee != controller_deprecated(), Error::<T>::ControllerDeprecated);
```

Comment:
You can do

```suggestion
			ensure!(
				(payee != {
					#[allow(deprecated)]
					RewardDestination::Controller
				}),
				Error::<T>::ControllerDeprecated
			);
```

Reply:
Oh yeah that looks clean, good call @georgepisaltu.